python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -863,9 +821,18 @@ def _process_auto_materialize_tick_generator(
                         status=TickStatus.STARTED,
                         timestamp=evaluation_time.timestamp(),
                         selector_id=instigator_selector_id,
-                        auto_materialize_evaluation_id=next_evaluation_id,
+                        # we base new auto_materialize_evaluation_ids on the tick id, which we need to
+                        # get from the response
+                        auto_materialize_evaluation_id=None,
+                    )
+                )
+                # set the new tick_id as the auto_materialize_evaluation_id
+                tick = tick._replace(
+                    tick_data=tick.tick_data._replace(
+                        auto_materialize_evaluation_id=override_evaluation_id or tick.tick_id
                     )
                 )
+                instance.update_tick(tick)
```

Comment:
Hmmm the fact that we are now doing two writes here in a relatively hot path where we used to do one is not ideal. Is there any world where we only write this field when it is an override and have this logic in the read path instead? If auto_materialize_evaluation_id is None, use the tick ID, otherwise use the override field?

Reply:
you are so right, I have no idea what past me was thinking -- updated the PR to avoid doing an additional write.